imports/api/employee.js

```diff
@@ -0,0 +1,234 @@
+import { Meteor } from "meteor/meteor";
+import { Employees, Tasks } from "./db";
+import EMPLOYEESAPI from "../constant/methods/employeesAPI";
+import { isAuthenticated } from "../util/authUtil";
+import { AuthError, EmployeeError } from "../constant/error";
+import { EmployeeMessage } from "../constant/message";
+import { removeElement, addToList } from "../util/arrayUtil";
+import { isAdmin, removeRelatedTasks } from "../util/employeeUtil";
+
+if (Meteor.isServer) {
+	Meteor.publish("employees", () => Employees.find());
+}
+
+export const insertEmployee = (
+	db,
+	_id,
+	employeeId,
+	firstName,
+	lastName,
+	role,
+) => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+	if (!isAdmin(_id, Employees)) {
+		throw new Meteor.Error(EmployeeError.ADMIN_NOT_FOUND);
+	}
+
+	return db.insert(
+		{
+			employeeId,
+			firstName,
+			lastName,
+			onBoard: new Date(),
+			// keep it null for now
+			managerId: null,
+			teamId: null,
+			tasksAssienedId: [],
+			tasksCreatedId: [],
+			tasksWatchingId: [],
+			requestsSentId: [],
+			requestReceivedId: [],
+			commentsId: [],
+			filesId: [],
+			employeesId: [],
+			individualsId: [],
+			teamsId: [],
+			// role should be one of"employer", "employee", "admin"?
+			role,
+		},
+		err => {
+			if (err) {
+				throw new Meteor.Error(EmployeeError.EMPLOYEE_INSERT_FAIL);
+			} else {
+				console.log(EmployeeMessage.EMPLOYEE_CREATED);
+			}
+		},
+	);
+};
+
+export const removeEmployee = (db, _id, employeeId) => {
+	if (!isAuthenticated()) {
+		throw new Meteor.Error(AuthError.NOT_AUTH);
+	}
+	if (!isAdmin(_id, Employees)) {
+		throw new Meteor.Error(EmployeeError.ADMIN_NOT_FOUND);
+	}
+	const employee = Employees.findOne({ employeeId });
+	const removedId = employee.employeeId;
+	if (!employee) {
+		throw new Meteor.Error(EmployeeError.EMPLOYEE_NOT_EXIST);
+	}
+	if (removedId !== this.employeeId) {
+		throw new Meteor.Error(AuthError.NO_PRIVILEGE);
+	}
+
+	return db.remove(
+		{
+			employeeId,
+		},
+		err => {
+			if (err) {
+				throw new Meteor.Error(EmployeeError.EMPLOYEE_REMOVE_FAIL);
+			} else {
+				let { tasksAssienedId } = employee;
+				tasksAssienedId = [...tasksAssienedId];
+				removeRelatedTasks(tasksAssienedId, employeeId, Tasks);
+				
+				let { tasksCreatedId } = employee;
+				tasksCreatedId = [...tasksCreatedId];
+				removeRelatedTasks(tasksCreatedId, employeeId, Tasks);
+
+				let { tasksWatchingId } = employee;
```

Comment:
Let's have a discussion here. If an employee is no longer with a company, we would only want to remove him/her from the task watch list since he/she no longer need to receive notification about changes. However, we wouldn't want to make any change to the tasks themselves. Imagine if an important task is assign to someone, just because the original creator got removed, doesn't mean the task is no longer valid. Same way with assigned tasks, in this case, we probably want to notify the creator to reassign it to someone else, but we would's want to delete it.

Reply:
I agree. I will use `removeWatcherFromTask` here to remove the person from the task. For the case where the task was assigned to this deleted employee, I think we need to add sth like `updateTaskAssignee` in `taskAPI`. Or we could use `assignTaskTo(db, _id, userId, assigneeId)`, but I don't know what I should put as `userId` in this case.

Reply:
We can just make it unassigned, maybe extend assignTaskTo to something like updateTaskAssignee to include the capability of making it unassigned.